core/types/transaction.go

```diff
@@ -651,11 +651,7 @@ func (tx *Transaction) AsMessage(s Signer, baseFee *big.Int) (Message, error) {
 		msg.gasPrice = math.BigMin(msg.gasPrice.Add(msg.gasTipCap, baseFee), msg.gasFeeCap)
 	}
 	var err error
-	if tx.Type() == ExternalTxType {
-		msg.from = common.ZERO
-	} else {
-		msg.from, err = Sender(s, tx)
-	}
+	msg.from, err = Sender(s, tx)
```

Comment:
Didn't this just get added a commit or two ago? Might need to edit the original commit

Reply:
I added it back for tx processing

Reply:
This is related to @gameofpointers's comment: https://github.com/dominant-strategies/go-quai/pull/470#discussion_r1010487728

Just correct the squash/split commits to get an atomic patch set.

tl;dr: if you add something in commit 1, but then remove it in commit 2, there's a good chance it shouldn't have been in commit 1 in the first place, and it just adds noise to the commit tree / PR

Reply:
I can no longer find the commit you're referring to. In the future please add a link to the commit in the comment/review

Reply:
Looks like you fixed it